### PR TITLE
UI tests for course overview page

### PR DESCRIPTION
**Pull request by Erin007**
A report came in through Zendesk of a teacher not being able to see any content on the course overview page. The issue more broadly affected any teacher who did not have any sections trying to view a course overview page. Fix went out in #31365. This follow-up includes lightweight regression tests for signed out, teacher without sections, student with sections and student without sections scenarios.   Teachers who manage sections viewing the course overview page is already covered in [teacher_homepage.feature](https://github.com/code-dot-org/code-dot-org/blob/6d0566fd42b9ca0b9e516f356c43e901847939e0/dashboard/test/ui/features/teacher_homepage.feature#L40).

### CODE REVIEW

**Comment by islemaster** [comment]
SauceLabs shows this error causing the drone failure:

image

File: dashboard/test/ui/features/course_overview.feature [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    Given I am a student
+    And I am on "http://studio.code.org/courses/csp-2019"
+    And I wait to see ".uitest-CourseScript"
+
+  Scenario: Viewing course overview as a teacher with no sections
+    Given I am a teacher
+    And I am on "http://studio.code.org/courses/csp-2019"
+    And I wait to see ".uitest-CourseScript"
+
+  Scenario: Viewing course overview as a student in a section
```
Comment (islemaster) [inline dashboard/test/ui/features/course_overview.feature:16]: This makes me want us to write a `Given I am a student in a section` step.

**Review by islemaster — APPROVED** [review]
Tests look great!  Maybe you need latest staging for drone to pass?

**Review by dmcavoy — APPROVED** [review]

**Comment by Erin007** [comment]
Today I (re?)learned that only some courses are available in UI tests: https://github.com/code-dot-org/code-dot-org/blob/b4a213885c0f23a72cb150fe0dd3d7f731cf7fbd/dashboard/lib/tasks/seed.rake#L166

**Pull request merged**